src/vector_windows.rs

```diff
@@ -21,20 +21,65 @@ pub mod service_control {
     };
 
     use crate::internal_events::{
-        WindowsServiceDoesNotExist, WindowsServiceInstall, WindowsServiceStart, WindowsServiceStop,
-        WindowsServiceUninstall,
+        WindowsServiceDoesNotExist, WindowsServiceInstall, WindowsServiceRestart,
+        WindowsServiceStart, WindowsServiceStop, WindowsServiceUninstall,
     };
     use crate::vector_windows::SERVICE_TYPE;
     use std::ffi::OsString;
+    use std::fmt;
     use std::time::Duration;
-    use std::{fmt, thread};
 
-    #[derive(Debug, Clone, PartialEq)]
+    #[derive(Debug)]
+    pub enum Error {
+        Service(windows_service::Error),
+        PollTimeout {
+            state: ServiceState,
+            expected_state: ServiceState,
+            timeout: Duration,
+        },
+    }
+
+    impl std::error::Error for Error {}
```

Comment:
We typically use `crate::Error` :)

Reply:
So `crate::Error` is a `pub type Error = Box<dyn std::error::Error + Send + Sync + 'static>;` which means that I can use anything that implements `std::error::Error`, or am I missing something ?

I wanted custom and clear error message so that I created my own `Error` type.

Reply:
I think this is a good reasoning. :)

Reply:
So you might want to use snafu like as shown in https://docs.rs/snafu/0.6.9/snafu/guide/the_macro/index.html.

If you give a look for `Snafu` in the vector codebase you can see how we often use it.

Reply:
Thanks for that, fixed in https://github.com/timberio/vector/pull/4408/commits/ae2dd39304d0e32d4515222b9cb2800aa19fd49b

Note that I had to use a custom implementation of `Display` for the `windows_service::Error` enum since I wanted to be able to extract the Winapi error instead of the generic "IO error in Winapi call" error message.

I'm not sure if it's the right way to do it, let me know if you have a better idea :)